### ISSUE TITLE
issue? tsuru key-add doesn't remove old key on overwrite

### ISSUE THREAD

**Issue by mtekel** [tsuru/tsuru#1249]
Suppose you do:
```
$ ssh-keygen -b 2048 -t rsa -f ~/.ssh/key_a -q -N ""
$ ssh-keygen -b 2048 -t rsa -f ~/.ssh/key_b -q -N ""
$ ssh-keygen -y -f ~/.ssh/key_a >~/.ssh/key_a.pub
$ ssh-keygen -y -f ~/.ssh/key_b >~/.ssh/key_b.pub

$ tsuru key-add mykey ~/.ssh/key_a.pub
Key "mykey" successfully added!
$ tsuru key-add mykey ~/.ssh/key_b.pub
Key "mykey" successfully added!
$ tsuru key-add mykey ~/.ssh/key_a.pub
Error: Key already exists.
```
When you now list keys with `tsuru key-list -n`, you will see the contents of key_b.pub. Nevertheless, key_a.pub is never removed from gandalf and does still keep working. Is this intended behaviour? Essentially, I'd expect that key to be removed/overwritten on update. I should then be able to overwrite/update the key back to original value. 

Also, having keys in gandalf, which do work but aren't listed in keys is a bit of security issue. You assume only listed keys work, but in reality there are some additional keys that can be still used to push apps.

**Comment by fsouza** [comment]
This is not the expected behavior. Thanks for reporting!

When reusing the name, we could issue an error message saying that the key is duplicate and asking whether the user wants to override it. The ``-y`` would apply as well.

**Comment by fsouza** [comment]
We can fix this by solving tsuru/gandalf#193 and tsuru/gandalf#194, along with some change in tsuru API code to:

- accept a force parameter in the add key handle
- when force is false and the key already exists (i.e. Gandalf returns 409 on key-add), report the error to the user
- when force is true and the key already exists (i.e. Gandalf returns 409 on key-add), call the update key endpoint (tsuru/gandalf#193).
 
In the client, we would implement a behavior similar to app-swap:

- if the user doesn't provide the ``-y`` flag, display the error message saying that the key is duplicate and ask whether the user wants to replace it
- if the user provide the -y flag, just send force=true to the tsuru API

**Comment by mtekel** [comment]
Thanks, that would fix the issue + add some nice key management options.
